connectors/src/connectors/google_drive/temporal/activities.ts

```diff
@@ -294,6 +299,8 @@ export async function incrementalSync(
       `Got changes.`
     );
     for (const change of changesRes.data.changes) {
+      await heartbeat();
```

Comment:
I think only this one is actually needed. The other heartbeat are in places that are quick to complete and happen only once per activity, so no real need to catch cancellation there.

It's not a big deal but it adds a little bit of complexity for not much I think. I would be in favor of removing them.

Reply:
Thanks, wasn't sure and what was long or not and wasn't that simple to test. Removing all the other ones 👍

Reply:
done